docker/cli
# Use goversioninfo to create Windows version info

Follow-up https://github.com/moby/moby/pull/42872

**- What I did**

Replace windres with native Go implementation to be able to create a syso file which contains Microsoft Windows Version Information and an icon using [goversioninfo](https://github.com/josephspurrier/goversioninfo).

**- How I did it**

Remove windres references and replace with goversioninfo. Also use `go:generate` for better integration and add legal copyright field.

**- How to verify it**

```shell
docker buildx bake --set "*.platform=windows/amd64"
#22 [linux/amd64 build 1/1] RUN --mount=type=bind,target=.,rw     --mount=type=cache,target=/root/.cache     --mount=from=dockercore/golang-cross:xx-sdk-extras,target=/xx-sdk,src=/xx-sdk     GO111MODULE=auto go install github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.3.0 &&     xx-go --wrap &&     TARGET=/out ./scripts/build/binary &&     xx-verify $([ "static" = "static" ] && echo "--static") /out/docker
#22 0.591 go: downloading github.com/josephspurrier/goversioninfo v1.3.0
#22 1.080 go: downloading github.com/akavel/rsrc v0.10.2
#22 2.167 Building static docker-windows-amd64.exe
#22 2.181 {
#22 2.181   "FixedFileInfo":
#22 2.181   {
#22 2.181     "FileVersion": {
#22 2.181       "Major": 20,
#22 2.181       "Minor": 10,
#22 2.181       "Patch": 2,
#22 2.181       "Build": 0
#22 2.181     },
#22 2.181     "FileFlagsMask": "3f",
#22 2.181     "FileFlags ": "00",
#22 2.181     "FileOS": "040004",
#22 2.181     "FileType": "01",
#22 2.181     "FileSubType": "00"
#22 2.181   },
#22 2.181   "StringFileInfo":
#22 2.181   {
#22 2.181     "Comments": "",
#22 2.181     "CompanyName": "Docker, Inc.",
#22 2.181     "FileDescription": "Docker Client",
#22 2.181     "FileVersion": "20.10.2.0",
#22 2.181     "InternalName": "",
#22 2.181     "LegalCopyright": "Copyright (c) 2013-2021 Docker Inc. All rights reserved.",
#22 2.181     "LegalTrademarks": "",
#22 2.181     "OriginalFilename": "docker-windows-amd64.exe",
#22 2.181     "PrivateBuild": "",
#22 2.181     "ProductName": "Docker Client",
#22 2.181     "ProductVersion": "20.10.2-262-g154ce5e58d.m",
#22 2.181     "SpecialBuild": ""
#22 2.181   },
#22 2.181   "VarFileInfo":
#22 2.181   {
#22 2.181     "Translation": {
#22 2.181       "LangID": "0409",
#22 2.181       "CharsetID": "04B0"
#22 2.181     }
#22 2.181   }
#22 2.181 }
#22 2.181 + go generate -v github.com/docker/cli/cmd/docker
#22 2.210 cmd/docker/docker.go
#22 2.210 cmd/docker/docker_test.go
#22 2.211 cmd/docker/docker_windows.go
#22 2.247 + go build -o /out/docker-windows-amd64.exe -tags  --ldflags '    -w          -X "github.com/docker/cli/cli/version.GitCommit=154ce5e58d"     -X "github.com/docker/cli/cli/version.BuildTime=2021-09-25T20:06:13Z"     -X "github.com/docker/cli/cli/version.Version=20.10.2-262-g154ce5e58d.m"      ' github.com/docker/cli/cmd/docker
#22 DONE 54.6s
```

should produce:

![image](https://user-images.githubusercontent.com/1951866/134784943-26fd080b-bc9e-4142-9c76-0e29c012f943.png)

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

cc @thaJeztah 

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>
